build.rs

```diff
@@ -0,0 +1,115 @@
+use crate::SqlType::{Idempotent, Migration};
+use askama::Template;
+use std::fs::File;
+use std::io::{Error, Read, Write};
+use std::path::{Path, PathBuf};
+use std::string::String;
+use std::{env, fs};
+
+#[derive(Template)]
+#[template(path = "migration-wrapper.sql", escape = "none")]
+struct MigrationWrapper<'a> {
+    filename: &'a str,
+    version: &'a str,
+    body: String,
+}
+
+#[derive(Template)]
+#[template(path = "idempotent-wrapper.sql", escape = "none")]
+struct IdempotentWrapper<'a> {
+    filename: &'a str,
+    body: String,
+}
+
+fn main() {
+    println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-changed=Cargo.lock");
+    println!("cargo:rerun-if-changed=migration");
+    println!("cargo:rerun-if-changed=templates");
+    println!("cargo:rerun-if-changed=migration-table.sql");
+
+    let mut sql = include_str!("migration-table.sql").to_string();
+    let rendered_sql = render_sql().expect("unable to render migrations");
+    sql.push_str(rendered_sql.as_str());
+
+    let mut out_path = env::current_dir().expect("current directory invalid");
```

Comment:
**Suggestion:** It's better to use `env::var("OUT_DIR")` which Cargo sets during compilation, and ensures that the output is placed in the compiler's output directory. You can then do things like `include!(concat!(env!("OUT_DIR"), "foo.txt"))` in your Rust code if needed.

If you want to source control the outputs, by placing the, relative to the crate directory that contains `Cargo.toml`, use `env::var("CARGO_MANIFEST_DIR")`, which is guaranteed to point to the correct location. There are a number of useful Cargo env vars you may find handy described [here](https://doc.rust-lang.org/cargo/reference/environment-variables.html)

Reply:
Yes, I am aware of `env::var("OUT_DIR")`. There are two considerations that led to the current implementation:

1. pgx doesn't (currently) support passing anything other than a string literal to `extension_sql_file`, so the input path has to be hard-coded. I have a PR open to enable support: https://github.com/zombodb/pgx/pull/407.

2. I don't really want to source control the output, but IMHO it would be nice to have the SQL which is output in this step in an "easy to find and view" location. `OUT_DIR` is a pain to find (there are multiple, with a hash appended, have to step through a wide tree of directories to find it). You're right, `CARGO_MANIFEST_DIR` is a better solution than `current_dir`, thanks for pointing that out!